lib/Signaling/Listener.php

```diff
@@ -158,6 +159,32 @@ protected static function registerExternalSignaling(IEventDispatcher $dispatcher
 		$dispatcher->addListener(Room::EVENT_AFTER_PARTICIPANT_TYPE_SET, $listener);
 		$dispatcher->addListener(Room::EVENT_AFTER_PARTICIPANT_PERMISSIONS_SET, $listener);
 
+		$dispatcher->addListener(Room::EVENT_AFTER_PERMISSIONS_SET, static function (RoomEvent $event) {
+			if (self::isUsingInternalSignaling()) {
+				return;
+			}
+
+			/** @var BackendNotifier $notifier */
+			$notifier = \OC::$server->get(BackendNotifier::class);
+
+			$sessionIds = [];
+
+			// Setting the room permissions resets the permissions of all
+			// participants, even those with custom attendee permissions.
+
+			/** @var ParticipantService $participantService */
+			$participantService = \OC::$server->get(ParticipantService::class);
+			$participants = $participantService->getSessionsAndParticipantsForRoom($event->getRoom());
+			foreach ($participants as $participant) {
+				$session = $participant->getSession();
+				if ($session) {
+					$sessionIds[] = $session->getSessionId();
+				}
+			}
```

Comment:
This does not scale. I think we need to remove it similarly to:
https://github.com/nextcloud/spreed/pull/6948

Reply:
I know, it is written in the pull request description :-P

However, does setting the default/call permission always reset the custom attendee permissions? Although [setting the default/call permissions through the endpoints resets the attendee permissions](https://github.com/nextcloud/spreed/blob/ac5d991f994b88360ab14a9ba6f31a86417c7727/tests/integration/features/conversation-2/set-permissions.feature#L92-L122) the RoomService API [allows modifying the permissions without resetting the custom permissions](https://github.com/nextcloud/spreed/blob/c756413574a9dfc86770e166cdd50b838ef89be4/lib/Service/RoomService.php#L158) (but I have not verified if it actually keeps the custom attendee permissions or if it refers to the custom room/call permissions and attendee permissions are always wiped).

If modifying the default/call permissions always resets the custom attendee permissions then a single event could be used here too, like with the end call, as it will have the same effect on every participant. However, if it is possible to modify the default/call permissions and still have custom attendee permissions then a single event would not work, as each participant could have a different set of final permissions :thinking:

Reply:
Both are possible:
* Without resetting:
https://github.com/nextcloud/spreed/blob/96074f7c957c3e66320890424a342f8a5cd6be8e/lib/Controller/RoomController.php#L1601-L1607
* With resetting:
https://github.com/nextcloud/spreed/blob/96074f7c957c3e66320890424a342f8a5cd6be8e/lib/Controller/RoomController.php#L1556-L1562

But both are not used in the current UI/clients. Maybe add a FIXME comment for now and then we remember it when we add a scaling test for this case.